plugins/TrollManagement/tests/TrollManagementTest.php

```diff
@@ -0,0 +1,153 @@
+<?php
+/**
+ * @author David Barbier<david.barbier@vanillaforums.com>
+ * @copyright 2009-2021 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace VanillaTests\TrollManagement;
+
+use UserModel;
+use RoleModel;
+use BanModel;
+use VanillaTests\SetupTraitsTrait;
+use VanillaTests\SiteTestCase;
+
+/**
+ * Class TrollManagementTest
+ */
+class TrollManagementTest extends SiteTestCase {
+    use SetupTraitsTrait;
+
+    /** @var UserModel */
+    protected $userModel;
+
+//    /** @var BanModel */
+//    protected $banModel;
+
+    /**
+     * Setup routine, run before the test class is instantiated.
+     */
+    public static function setupBeforeClass(): void {
+        self::$addons = ['vanilla', 'trollmanagement'];
+
+        parent::setupBeforeClass();
+    }
+
+    /**
+     * Tests the automatic assignment of the "applicant" status to a new user registration if the maximum amount of
+     * user accounts sharing the same fingerprint is reached.
+     */
+    public function testRegisterApplicant(): void {
+        /** @var \Gdn_Configuration $configuration */
+        $configuration = static::container()->get('Config');
+
+        $configuration->set('TrollManagement.PerFingerPrint.Enabled', true);
+        $configuration->set('TrollManagement.PerFingerPrint.MaxUserAccounts', 3);
+
+        // Ensure all future registered dummy users uses the same fingerprint.
+        $_COOKIE['__vnf'] = 'THISISAFAKEFINGERPRINT';
+
+        // Create 3 dummy accounts. (The third should be automatically set as "applicant")
+        $importedUsers[] = $this->insertDummyUser();
+        $importedUsers[] = $this->insertDummyUser();
+        $importedUsers[] = $this->insertDummyUser();
+
+        // We pull the associated user's roles.
+        foreach ($importedUsers as $importedUser) {
+            $importedUsersRolesIDs[] = $this->userModel->getRoleIDs($importedUser['UserID']);
+        }
+
+        // The FIRST dummy user account is NOT an applicant
+        $this->assertNotContains(RoleModel::APPLICANT_ID, $importedUsersRolesIDs['0']);
+        // The SECOND dummy user account is NOT an applicant
+        $this->assertNotContains(RoleModel::APPLICANT_ID, $importedUsersRolesIDs['1']);
+        // The THIRD dummy user account should an applicant
+        $this->assertContains(RoleModel::APPLICANT_ID, $importedUsersRolesIDs['2']);
+    }
+
+    /**
+     * Tests the automatic assignment of the "applicant" status to EVERY new user registration.
+     */
+    public function testRegisterAllApplicants(): void {
```

Comment:
If we're adding a test here, we're essentially saying this is expected and intended behavior. Is adding "make every new registration an applicant" an intended behavior of the Troll Management addon? As discussed, we already have the applicant registration method to make all new registrations come in as applications. Instead, I would recommend enforcing "MaxUserAccounts" be no less than 1.

Reply:
Agreed. I removed the test & added a bessy test that checks for validation in the backend.